authority/provisioner/x5c_test.go

```diff
@@ -456,13 +458,36 @@ func TestX5C_AuthorizeSign(t *testing.T) {
 				sans:  []string{"127.0.0.1", "foo", "max@smallstep.com"},
 			}
 		},
+		"ok/cnf": func(t *testing.T) test {
+			p, err := generateX5C(nil)
+			assert.FatalError(t, err)
+
+			x5c := make([]string, len(certs))
+			for i, cert := range certs {
+				x5c[i] = base64.StdEncoding.EncodeToString(cert.Raw)
+			}
+			extraHeaders := map[string]any{"x5c": x5c}
+			extraClaims := map[string]any{
+				"sans": []string{"127.0.0.1", "foo", "max@smallstep.com"},
+				"cnf":  map[string]any{"x5rt#S256": "fingerprint"},
+			}
+
+			tok, err := generateCustomToken("foo", p.GetName(), testAudiences.Sign[0], jwk, extraHeaders, extraClaims)
+			assert.FatalError(t, err)
+			return test{
+				p:           p,
+				token:       tok,
+				sans:        []string{"127.0.0.1", "foo", "max@smallstep.com"},
+				fingerprint: "fingerprint",
+			}
+		},
 	}
 	for name, tt := range tests {
 		t.Run(name, func(t *testing.T) {
 			tc := tt(t)
 			ctx := NewContextWithMethod(context.Background(), SignIdentityMethod)
 			if opts, err := tc.p.AuthorizeSign(ctx, tc.token); err != nil {
-				if assert.NotNil(t, tc.err) {
+				if assert.NotNil(t, tc.err, err.Error()) {
```

Comment:
A bit of a weird test case. The `tc.err` is the expected error, and an assertion on that is unexpected. It was already there, but I don't think this is the right test logic now.

Reply:
It works as expected, showing the `err.Error()` if the assertion fails, but with ad70982c I've changed this file to use testify's assert and require packages.